monai/data/dataset.py

```diff
@@ -1173,7 +1170,7 @@ def shutdown(self):
         # wait until replace mgr is done the current round
         while not self._try_shutdown():
             time.sleep(0.01)
-        self._replace_mgr.join()
+        self._replace_mgr.join(300)
```

Comment:
May I know what issue this `300` change to solve?

Thanks.

Reply:
Yes, the 'join' blocks the entire workflow https://github.com/Project-MONAI/MONAI/issues/5660 I've seen it randomly many times, so trying to add a timeout to avoid waiting indefinitely. It's safe to do so because we always check if there is a thread alive and always create a new one with daemon=True.